src/bcd.c

```diff
@@ -239,14 +239,30 @@ bcd_child_exit(int e)
 }
 
 #ifdef __linux__
-#ifndef gettid
+#if !defined(__GLIBC_PREREQ) || !__GLIBC_PREREQ(2, 30)
 static pid_t
 gettid(void)
 {
 
 	return syscall(__NR_gettid);
 }
-#endif /* !gettid */
+#endif /* !defined(__GLIBC_PREREQ) || !__GLIBC_PREREQ(2, 30) */
+#endif /* __linux__ */
+
+#ifdef __linux__
+static size_t
+strlcpy(char *dst, const char *src, size_t n)
+{
+	size_t len = strlen(src);
+
+	if (n > 0) {
```

Comment:
`size_t` is defined to be unsigned, so this `if` is not necessary: 

http://www.cplusplus.com/reference/cstring/size_t/

There is the `ssize_t` type where negative values are desired (for example in a function that returns -1 on error, otherwise a size).

Reply:
The condition is needed and I find it more readable than `if (n != 0) {`.

Reply:
Oh right, `n` can be `0`.